metapackages/auto-instrumentations-node/src/utils.ts

```diff
@@ -136,3 +161,63 @@ export function getNodeAutoInstrumentations(
 
   return instrumentations;
 }
+
+export function getResourceDetectorsFromEnv(): Array<Detector | DetectorSync> {
+  const resourceDetectorsFromEnv =
+    process.env.OTEL_NODE_RESOURCE_DETECTORS?.split(',');
+
+  if (
+    resourceDetectorsFromEnv == null ||
+    resourceDetectorsFromEnv.includes('all')
```

Comment:
can be simplified to 
```suggestion
    resourceDetectorsFromEnv?.includes('all')
```

Reply:
Simplifying it will cause `resourceDetectorsFromEnv.forEach` a couple of lines below to throw an error since it will think it could still be undefined at that point.